remove version constrain on cryptography. obove of the current versio…

jfdelgad opened the pull request:
…n the library runs without problems in mac with M1 processors

Fixes #1034 .
Installing ocean.py in Mac with M1 processor fail due to issues with Cryptography #1034

Changes proposed in this PR:

- upgrade to a newer version of Cryptography
- 
-

trentmc commented:
This issue was initially reported in #887.

The fix was: in installation instructions in data-nfts-and-datatokens-flow.md README, have:

```
Known issues:

for M1 processors, coincurve and cryptography installation may fail due to dependency/compilation issues. It is recommended to install them individually, e.g. pip3 install coincurve && pip3 install cryptography
```

However, the proposed fix here is simpler, because it doesn't require user intervention. It just uses the latest version of the library. Nice! 

Does it work for both libs `coincurve` and for `cryptography`?

If yes, great! Please ensure that both are updated in setup.py. 

And, please update the instructions in the README as appropriate. (Ideally it works for both libs; then we can completely remove the sentence "for M1 processors..."

trentmc reviewed (changes requested):
Please see my comment: see if it can works for both libs: coincurve & cryptography. And, update the README as appropriate.

jfdelgad commented:
> This issue was initially reported in #887.
> 
> The fix was: in installation instructions in data-nfts-and-datatokens-flow.md README, have:
> 
> ```
> Known issues:
> 
> for M1 processors, coincurve and cryptography installation may fail due to dependency/compilation issues. It is recommended to install them individually, e.g. pip3 install coincurve && pip3 install cryptography
> ```
> 
> However, the proposed fix here is simpler, because it doesn't require user intervention. It just uses the latest version of the library. Nice!
> 
> Does it work for both libs `coincurve` and for `cryptography`?
> 
> If yes, great! Please ensure that both are updated in setup.py.
> 
> And, please update the instructions in the README as appropriate. (Ideally it works for both libs; then we can completely remove the sentence "for M1 processors..."

Coincurve failed was because an outdated library in my local machine, after updating it coincurve installation went through. So I think we can safely reove the M1 message in the read me.

AnaLoznianu commented:
@jfdelgad Great to see this PR :) 
If your changes fixes the problem, can you please update also: https://github.com/oceanprotocol/ocean.py/blob/main/READMEs/data-nfts-and-datatokens-flow.md. We have that mention of m1 issues. Thank you!

mariacarmina commented:
@jfdelgad, can we also update the READMEs as well? Also as @trentmc said, can we check if `coincurve` can also be updated?

jfdelgad commented:
> @jfdelgad, can we also update the READMEs as well? Also as @trentmc said, can we check if `coincurve` can also be updated?

The coincurve issue was related to libtools, updating it makes the issue to go away, it is not related to the version of coincurve.

jfdelgad commented:
> @jfdelgad, can we also update the READMEs as well? Also as @trentmc said, can we check if `coincurve` can also be updated?

updated the readme, removed the line that suggested to install the libraries first. that was actually not a valid solution for me.

calina-c commented:
@jfdelgad can you please fix the conflicts and apply the same change in a different PR for `main-brownie`? I reverted the main branch to its pre-brownie stage, and that's where I think the conflicts are coming from. It's also a measure of precaution from my part to double everything into `main-brownie` as well, in order to help testing and to have a "complete" functionality there as well.

jfdelgad commented:
> @jfdelgad can you please fix the conflicts and apply the same change in a different PR for `main-brownie`? I reverted the main branch to its pre-brownie stage, and that's where I think the conflicts are coming from. It's also a measure of precaution from my part to double everything into `main-brownie` as well, in order to help testing and to have a "complete" functionality there as well.

This was already merged to the main branch in another PR, I will close it.
